src/services/users/controllers/deleteAccount.ts

```diff
@@ -4,21 +4,123 @@ import { ObjectId } from 'mongodb';
 import { IRequest } from '../../../ts/interfaces/global.interfaces';
 import service from '../../../helpers/services';
 import { action } from '../../../helpers/accessControl/accessList';
+import { IUser } from '../../../ts/interfaces/db.interfaces';
+
+const checkIfUserIsContactStructure =
+  (app: Application, req: IRequest) => async (idUser: string) => {
+    const user = await app
+      .service(service.users)
+      .Model.accessibleBy(req.ability, action.read)
+      .findOne({ _id: new ObjectId(idUser) })
+      .select({ name: 1, _id: 0 });
+    const structure = await app
+      .service(service.structures)
+      .Model.accessibleBy(req.ability, action.read)
+      .countDocuments({
+        'contact.email': user.name,
+      });
+    if (structure > 0) {
+      return true;
+    }
+    return false;
+  };
 
 const deleteAccount =
   (app: Application) => async (req: IRequest, res: Response) => {
     const idUser = req.params.id;
+    const { actionSuppression } = req.query;
+    let user: undefined | IUser;
     try {
-      // xxx pour le moment on n'autorise que les admins à supprimer un user
-      // à voir : suppression d'un rôle et non du user complet si multi rôle
-      await app
-        .service(service.users)
-        .Model.accessibleBy(req.ability, action.delete)
-        .deleteOne({
-          _id: new ObjectId(idUser),
-          roles: { $all: ['admin'] },
-        });
-      res.status(200).json({ deleteSuccess: true, idUser });
+      if (actionSuppression === 'tous') {
+        const isContactStructure = await checkIfUserIsContactStructure(
+          app,
+          req,
+        )(idUser);
+        if (isContactStructure) {
+          return res.status(409).json({
+            message: 'le gestionnaire ne peut pas être supprimé',
+          });
+        }
+        await app
+          .service(service.users)
+          .Model.accessibleBy(req.ability, action.delete)
+          .deleteOne({ _id: new ObjectId(idUser) });
+      } else {
+        let query = {};
+        switch (actionSuppression) {
+          case 'grandReseau':
+            query = {
+              $unset: {
+                reseau: '',
+              },
+              $pull: {
+                roles: 'grandReseau',
+              },
+            };
+            break;
+          case 'structure' || 'structure_coop':
```

Comment:
Syntaxe non valide dans un case mais on enverra depuis le front que 'structure'
```suggestion
          case 'structure':
```